core/src/main/java/org/infinispan/interceptors/locking/PessimisticLockingInterceptor.java

```diff
@@ -77,6 +82,26 @@ protected final Object visitDataReadCommand(InvocationContext ctx, DataCommand c
    }
 
    @Override
+   public Object visitGetAllCommand(InvocationContext ctx, GetAllCommand command) throws Throwable {
+      try {
+         if (ctx.isInTxScope() && command.hasFlag(Flag.FORCE_WRITE_LOCK) && !hasSkipLocking(command)) {
+            acquireRemoteIfNeeded(ctx, command.getKeys(), command);
+            final TxInvocationContext txContext = (TxInvocationContext) ctx;
+            long lockTimeout = getLockAcquisitionTimeout(command, false);
+            for (Object key : command.getKeys()) {
+               lockAndRegisterBackupLock(txContext, key, lockTimeout, false);
+            }
+         }
+         return invokeNextInterceptor(ctx, command);
+      } catch (Throwable t) {
+         releaseLocksOnFailureBeforePrepare(ctx);
+         throw t;
+      } finally {
+         if (!ctx.isInTxScope()) lockManager.unlockAll(ctx);
```

Comment:
To be honest I thought I had removed this.  It was there due to L1 locking, but I disabled updates to L1 with putMap.